velox/vector/BaseVector.h

```diff
@@ -635,8 +635,24 @@ class BaseVector {
     return left == right || right == TypeKind::UNKNOWN;
   }
 
+  /// Returns a brief summary of the vector that includes encoding, type, number
+  /// of rows and number of nulls.
+  ///
+  /// For example,
+  ///     [FLAT INTEGER: 3 elements, no nulls]
+  ///     [DICTIONARY INTEGER: 5 elements, 1 nulls]
+  ///     [CONSTANT INTEGER: 100 elements, no nulls]
   virtual std::string toString() const;
 
+  /// Returns a brief summary of all the layers of encodings of this vector.
+  ///
+  /// For example,
+  ///
+  ///     [DICTIONARY INTEGER: 5 elements, no nulls], [FLAT INTEGER: 10
+  ///     elements, no nulls]
+  std::string toString(bool recursive) const;
```

Comment:
nit: do we want to combine the two with default parameter to false?
```
std::string toString(bool recursive = false) const;
```

Reply:
I did that initially, but toString() is a virtual method and ConstantVector provides custom implementation. Hence, I thought that having a separate non-virtual method would be better. I'm happy to change this though. Let me know your thoughts.